molpipeline/estimators/chemprop/abstract.py

```diff
@@ -95,3 +142,106 @@ def fit(
         )
         self.lightning_trainer.fit(self.model, training_data)
         return self
+
+    def set_params(self, **params: Any) -> Self:
+        """Set the parameters of the model.
+
+        Note
+        ----
+        Parameters for the trainer and the checkpoint callback are filtered out and added as attributes of the model.
+        This is done due to incompatibility with the `get_params` method.
+
+        Parameters
+        ----------
+        **params: Any
+            The parameters to set.
+
+        Returns
+        -------
+        Self
+            The model with the new parameters.
+        """
+        params, trainer_params = self._filter_params_trainer(params)
+        params, model_ckpoint_params = self._filter_params_callback(params)
+        self.trainer_params.update(trainer_params)
+        self.model_ckpoint_params.update(model_ckpoint_params)
+        super().set_params(**params)
+        self._update_trainer()
+        return self
+
+    def get_params(self, deep: bool = False) -> dict[str, Any]:
+        """Get the parameters of the model.
+
+        Parameters
+        ----------
+        deep : bool, optional (default=False)
+            Whether to get the parameters of the model.
+
+        Returns
+        -------
+        dict[str, Any]
+            The parameters of the model.
+        """
+        params = super().get_params(deep)
+        for name, value in self.trainer_params.items():
+            params[f"lightning_trainer__{name}"] = value
+        for name, value in self.model_ckpoint_params.items():
+            params[f"callback_modelckpt__{name}"] = value
+        # set to none as the trainer is created from the parameters
+        # Otherwise, the sklearn clone will fail as the trainer is updated by replacing the object
```

Comment:
Maybe write in the comment that for the lightning_trainer *deep* is not optional. We can't give out (return) the object but we must return the params only.

Reply:
Done